src/gwt/src/org/rstudio/studio/client/workbench/views/environment/EnvironmentPane.java

```diff
@@ -415,8 +415,17 @@ private void executeFunctionForObject(String function, String objectName)
    private Widget createImportMenu()
    {
       ToolbarPopupMenu menu = new ToolbarPopupMenu();
-      menu.addItem(commands_.importDatasetFromFile().createMenuItem(false));
-      menu.addItem(commands_.importDatasetFromURL().createMenuItem(false));
+      
+      if (prefs_.useDataImport().getValue())
+      {
+         menu.addItem(commands_.importDatasetFromCSV().createMenuItem(false));
```

Comment:
Once you have the remove logic in place you can just add all of these items to the menu and the removed ones won't be displayed.